modules/dnn/src/layers/batch_norm_layer.cpp

```diff
@@ -272,6 +277,26 @@ class BatchNormLayerImpl CV_FINAL : public BatchNormLayer
         inputs_arr.getMatVector(inputs);
         outputs_arr.getMatVector(outputs);
 
+        if (inputs[0].dims <= 1) { // Handling for 0D and 1D
+            Mat &inpBlob = inputs[0];
+            Mat &outBlob = outputs[0];
+
+            if (inputs[0].dims == 0) {  // Scalar input
+                float inpVal = inputs[0].at<float>();
+                float outVal = inpVal * weights_.at<float>(0) + bias_.at<float>(0);
+                outBlob.at<float>(0) = outVal;
+            } else {  // Vector input
+                for (int i = 0; i < inpBlob.size[0]; i++) {
+                    float inpVal = inpBlob.at<float>(i);
+                    float outVal = inpVal * weights_.at<float>(i) + bias_.at<float>(i);
+                    outBlob.at<float>(i) = outVal;
+                }
```

Comment:
you can use cv::multiply and cv::add instead of loop. Or, even simpler outBlob = inpBlob * weights_ + bias_, if they have the same type and compatible shapes.